bftengine/src/bftengine/SigManager.cpp

```diff
@@ -88,23 +86,17 @@ SigManager* SigManager::init(ReplicaId myId,
                              uint16_t numRoReplicas,
                              uint16_t numOfClientProxies,
                              uint16_t numOfExternalClients) {
-  if (instance_) {
-    // allow re-configuration..
-    LOG_INFO(GL, "SigManager already exist, re-creating..");
-    delete instance_;
-    instance_ = nullptr;
-  }
-  instance_ = initImpl(myId,
-                       mySigPrivateKey,
-                       publicKeysOfReplicas,
-                       replicasKeysFormat,
-                       publicKeysOfClients,
-                       clientsKeysFormat,
-                       numReplicas,
-                       numRoReplicas,
-                       numOfClientProxies,
-                       numOfExternalClients);
-  return instance_;
+  SigManager* sm = initImpl(myId,
+                            mySigPrivateKey,
+                            publicKeysOfReplicas,
+                            replicasKeysFormat,
+                            publicKeysOfClients,
+                            clientsKeysFormat,
+                            numReplicas,
+                            numRoReplicas,
+                            numOfClientProxies,
+                            numOfExternalClients);
+  return SigManager::getInstance(sm);
```

Comment:
We use the `instance()` everywhere

Reply:
@toly-kournik , I will rename it in another dedicated PR.
I suggest keeping this name as-is for now, this modification is urgent.